src/sentry/tasks/integrations/github/open_pr_comment.py

```diff
@@ -352,34 +364,39 @@ def open_pr_comment_workflow(pr_id: int) -> None:
     # CREATING THE COMMENT
 
     # fetch the files in the PR and determine if it is safe to comment
-    safe_to_comment, pr_files = safe_for_comment(
-        gh_client=client, repository=repo, pull_request=pull_request
-    )
+    pr_files = safe_for_comment(gh_client=client, repository=repo, pull_request=pull_request)
 
-    if not safe_to_comment:
-        logger.info("github.open_pr_comment.not_safe_for_comment")
+    if len(pr_files) == 0:
+        logger.info(
+            "github.open_pr_comment.not_safe_for_comment", extra={"file_count": len(pr_files)}
+        )
         metrics.incr(
             OPEN_PR_METRICS_BASE.format(key="error"),
             tags={"type": "unsafe_for_comment"},
         )
         return
 
-    # TODO(cathy): return patches
-    pr_filenames, _ = get_pr_filenames_and_patches(pr_files)
+    pr_filenames, patches = get_pr_filenames_and_patches(pr_files)
 
     issue_table_contents = {}
     top_issues_per_file = []
 
     # fetch issues related to the files
-    # TODO(cathy): include fetching function names
-    for pr_filename in pr_filenames:
+    for i, pr_filename in enumerate(pr_filenames):
         projects, sentry_filenames = get_projects_and_filenames_from_source_file(
             org_id, pr_filename
         )
         if not len(projects) or not len(sentry_filenames):
             continue
 
-        top_issues = get_top_5_issues_by_count_for_file(list(projects), list(sentry_filenames))
+        function_names = get_file_functions(patches[i])
```

Comment:
Can we be confident that `patches` will have all the indexes/len of `pf_filenames`? Can we do an assertion before starting this for loop and return a specific error/log if not?

Reply:
The reason I ask to have this, is because if someone changes the logic in `get_pr_filenames_and_patches` without realizing the dependency of a 1:1, this will crash unexpectedly as a by product

Reply:
i made a dataclass -- thanks for the suggestion!

Reply:
For future reference, we talked about this offline a little: https://sentry.slack.com/archives/C04HEFTMQUC/p1703092269575069?thread_ts=1703089800.779749&cid=C04HEFTMQUC